### DISCUSSION TITLE
[Question] Register command pop usage ui

### DISCUSSION THREAD

**Discussion by CppCXY** [redhat-developer/lsp4ij#363] [Q&A] [answered]
![image](https://github.com/redhat-developer/lsp4ij/assets/40318218/f0136b5d-0c46-4787-9924-82da343536cb)
My own [language server](https://github.com/CppCXY/EmmyLuaAnalyzer) supports the CodeLens feature, but it needs to use the client's command to implement the click to pop up usages. How can I register such a command and pop up the usage UI?

**Comment by angelozerr** [comment, marked as answer]
See doc at https://github.com/redhat-developer/lsp4ij/blob/main/docs%2FDeveloperGuide.md#lsp-commands

**Reply by fbricon** [reply]
See example of client specific commands : https://github.com/redhat-developer/intellij-quarkus/blob/main/src/main/resources/META-INF/plugin.xml#L680-L694

**Reply by angelozerr** [reply]
@fbricon I think we should improve documentation to give a sample piece of code how to implement LSPCommand and provide the link you have set.

**Comment by angelozerr** [comment]
I close this discussion in favor of https://github.com/redhat-developer/lsp4ij/issues/553
